packages/calcite-components/src/components/list/list.tsx

```diff
@@ -813,15 +819,27 @@ export class List
     this.updateFilteredData();
   }
 
-  private getItemData(): ItemData {
+  private getItemData(): ItemData[] {
     return this.listItems.map((item) => ({
       label: item.label,
       description: item.description,
       metadata: item.metadata,
+      heading: this.getGroupHeading(item),
       el: item,
     }));
   }
 
+  private getGroupHeading(item: ListItem["el"]): string {
+    let heading = "";
+    this.listItemGroups.forEach((group) => {
+      if (group.contains(item)) {
+        heading += group.heading;
```

Comment:
We could also store heading of ancestor as Array object.

Leaning towards string concatenation for performance gains with `filter` method.

Reply:
Yeah I was thinking array because string concat would match two words from different headings combined. Can we change to use an array here?